### PR TITLE
SOLR-15897 Remove <jmx/> from test solrconfig.xml

### PR DESCRIPTION
**Pull request by epugh**
https://issues.apache.org/jira/browse/SOLR-15897

# Description

Unit tests have deprecated <jmx/> tags in solrconfig.xml

# Solution

Removed <jmx/> from test solrconfigs.xml plus a configset related to jmx that is not used.

# Tests

Re ran the tests.

# Checklist

Please review the following and check all that apply:

- [X ] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [ X] I have created a Jira issue and added the issue ID to my pull request title.
- [ X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ X] I have developed this patch against the `main` branch.
- [ X] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/CHANGES.txt [hunk 1]
```diff
@@ -427,6 +427,8 @@ Other Changes
 
 * SOLR-15807: New LogListener class for tests to use to make assertions about what Log messages should or should not be produced by a test (hossman)
 
+* SOLR-15897: Remove <jmx/> from all unit test solrconfig.xml files. (Eric Pugh)
+
```
Comment (cpoerschke) [inline solr/CHANGES.txt:431]: Addition here presumes backporting to `branch_9_0_0` branch, I have no views for or against that, just noting it as part of checking the entry is in the correct section :-)

**Review by cpoerschke — APPROVED** [review]

File: solr/CHANGES.txt (hunk 1 above)
Comment (epugh) [reply solr/CHANGES.txt:431]: thanks!  I think I didn't quite understand how the branches worked ;-)

Comment (epugh) [reply solr/CHANGES.txt:431]: Okay, I think I just did the back port...

**Comment by epugh** [comment]
@cpoerschke can you review now?  Did I get the CHANGES.txt right?  I want this to not go in Solr 9.0.0.  I would like it to eventually go in Solr 9.1.0 however.

**Comment by cpoerschke** [comment]
> @cpoerschke can you review now? Did I get the CHANGES.txt right? I want this to not go in Solr 9.0.0. I would like it to eventually go in Solr 9.1.0 however.

Interesting, there's no 9.1.0 section as yet in CHANGES.txt on `main` (and so the location right now is the 10.0.0 section actually).

@janhoy would the 9.1.0 section be added as part of the 9.0.0 release process (at some point) or can/should developers add it as/when the need arises? Somehow I'm thinking the former but maybe things are different for a major release etc.

**Comment by cpoerschke** [comment]
> @janhoy would the 9.1.0 section be added as part of the 9.0.0 release process (at some point) or can/should developers add it as/when the need arises? Somehow I'm thinking the former but maybe things are different for a major release etc.

Partly answering own question: `branch_9x` has https://github.com/apache/solr/commit/e9a47a4b9a1d41bb9c1991b1c1cae5170618dd51#diff-cdd1ae1b418ea74ca689c6e68e61f4e5b4c6f77f6408a4e6cf76ed8421b24d09

**Pull request merged**